src/checkers/flymake-collection-statix.el

```diff
@@ -0,0 +1,134 @@
+;;; flymake-collection-statix.el --- Statix diagnostic function -*- lexical-binding: t -*-
+
+;; Copyright (c) 2022 Valeriy Litkovskyy
+
+;; Permission is hereby granted, free of charge, to any person obtaining a copy
+;; of this software and associated documentation files (the "Software"), to deal
+;; in the Software without restriction, including without limitation the rights
+;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+;; copies of the Software, and to permit persons to whom the Software is
+;; furnished to do so, subject to the following conditions:
+
+;; The above copyright notice and this permission notice shall be included in all
+;; copies or substantial portions of the Software.
+
+;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+;; SOFTWARE.
+
+;;; Commentary:
+
+;; This file provides a flymake backend that lets the user analyze nix code
+;; using statix.
+
+;;;; Tips
+
+;; + You can customize `flymake-collection-statix-command' to your liking.
+
+;; + Some warnings have "{fix}" suffix.  They can be fixed by calling
+;;   `flymake-collection-statix-fix'.
+
+;;;; Credits
+
+;; This diagnostic would not have been possible without the statix[1] program.
+
+;;  [1] https://github.com/nerdypepper/statix
+
+;;; Code:
+
+;;;; Requirements
+
+(require 'flymake)
+(require 'flymake-collection)
+
+(eval-when-compile
+  (require 'flymake-collection-define))
+
+;;;; Customization
+
+(defcustom flymake-collection-statix-command
+  '("statix" "check" "--format" "json" "--stdin")
+  "Statix command.
+Make sure that it accepts json on stdin."
+  :type '(repeat :tag "Command" (string :tag "Part"))
+  :group 'flymake-collection)
+
+;;;; Commands
+
+(defun flymake-collection-statix-fix ()
```

Comment:
This is interesting, but perhaps would be better as a more general purpose command that can be applied for other checkers as well. Please let me consider where the best interface for this would be :thinking:.

Reply:
Indeed, many checkers offer fixes for the region of code (like eslint).

I think if **flymake-collection** had a general interface for fixes, it would be a great selling point for the package. After all, I decided to move my [package](https://github.com/xFA25E/flymake-statix) here, because I don't think that maintaining a lot of separete flymakes is the way to go (it is one of the selling points of flycheck).

The only problem, at least to me, is that I'm not sure whether flymake should deal with fixes or it should remain a linter.

If flymake should handle fixes, then, maybe, it should be a general interface in flymake core? But, flymake-collection has already o nice mechanism for loading checkers that is not in the core (and I think it should be). I'm not sure, what do you think?

Reply:
> The only problem, at least to me, is that I'm not sure if flymake should handle fixes or if it should remain a linter.

I think it makes sense to add features we'd like in standard flymake to `flymake-collection`. If they prove valuable we can eventually port them to flymake core as well (although the license of this project may complicate that if we want to just contribute code). Either way since the fixes depend on the checkers at least for now I don't see much value in separating them.

> Please let me consider where the best interface for this would be thinking

I still haven't reached a consensus :upside_down_face:. Would you consider removing this functionality from this PR and adding it back in a follow up PR? I'd rather not hold back the checker definition due to a related feature.